### ISSUE TITLE
Ivy: fullTemplateTypeCheck errors with NgIterable

### ISSUE THREAD

**Issue by cexbrayat** [angular/angular#33527]
# 🐞 bug report

### Affected Package

The issue is caused by package @angular/compiler

### Is this a regression?

This is introduced by Ivy.

### Description

A component with a template looking like:

```html
<div *ngFor="let user of users | slice: 0:2 as total">
  {{ total.length }}: {{ user.name }}
</div>
```

`ng build --prod` with `fullTemplateTypeCheck: true` throws, as it thinks `length` is not a property of `total`.

## 🔬 Minimal Reproduction

https://github.com/cexbrayat/ng-iterable

## 🔥 Exception or Error
<pre><code>
ERROR in src/app/app.component.html:2:6 - error TS2339: Property 'length' does not exist on type 'NgIterable<any>'.
  Property 'length' does not exist on type 'Iterable<any>'.

2   {{ total.length }}: {{ user.name }}
       ~~~~~~~~~~~~~

  src/app/app.component.ts:5:16
    5   templateUrl: './app.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component AppComponent.
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.0.0-rc.0
Node: 12.12.0
OS: darwin x64
Angular: 9.0.0-rc.0
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.0
@angular-devkit/build-angular     0.900.0-rc.0
@angular-devkit/build-optimizer   0.900.0-rc.0
@angular-devkit/build-webpack     0.900.0-rc.0
@angular-devkit/core              9.0.0-rc.0
@angular-devkit/schematics        9.0.0-rc.0
@ngtools/webpack                  9.0.0-rc.0
@schematics/angular               9.0.0-rc.0
@schematics/update                0.900.0-rc.0
rxjs                              6.5.3
typescript                        3.6.4
webpack                           4.41.2
</code></pre>

**Anything else relevant?**

This is using ng and CLI `9.0.0-rc.0` and `fTTC`, but *not* using `strictTemplates`.

**Comment by cexbrayat** [comment]
@alxhub I'm still encoutering the issue using `rc.3` which should include the fix.
I updated the repro https://github.com/cexbrayat/ng-iterable

**Comment by JoostK** [comment]
Closing as fixed in #33997.
